Add Grafana how-to guide (#59)
crate/cloud-howtos: docs/visualize-data-with-grafana.rst

.. _visualize-data-with-grafana:

===========================
Visualize data with Grafana
===========================

`Grafana`_ is an open-source tool that helps you build real-time dashboards, 
graphs, and all sorts of data visualizations. It is the perfect complement 
to CrateDB, which is purpose-built for monitoring large volumes of machine 
data in real-time.

For the purposes of this guide, it is assumed that you
have a cluster up and running and can access the Console. If not, please refer
to the :ref:`tutorial on how to deploy a cluster for the first time
<cloud-tutorials:cluster-deployment>`.

.. rubric:: Table of contents

.. contents::
   :local:

.. _grafana-load-dataset:

Load a sample dataset
=====================

To visualize data with Grafana, a dataset is needed first. In this sample, demo

> data is added directly via the CrateDB Cloud Console. To import the data go to the Overview page of your deployed cluster. 

Reviewer: Please wrap conforming to style guide

Author: Ah, right.. good catch. I was playing with auto-wrapper throughout work on this. Must have left it on soft-wrap.

.. image:: _assets/img/grafana-overview.png
   :alt: Cloud Console cluster overview

Once on the Overview page, click on the *Learn how to import data to your
cluster* link in the bottom part of the Console. A window with 2 SQL
statements will appear. The first of them creates a table that will host the data

> from NYC Taxi & Limousine Commission which is used in this example. The second statement imports the data into the table created in the first step. These statements must be executed in order. First "1. Create the table" and then "2. Import the data".

Reviewer: Please wrap conforming to style guide

   :alt: Creating table in Admin UI

After executing the second SQL statement, the "nyc_taxi" table will be
populated with data. Depending on your cluster configuration this can take
around 40 minutes.

.. _grafana-install:

Install Grafana
===============

To install Grafana locally, refer to the `Grafana documentation`_. In this
guide, local installation is used but you can also use Grafana cloud
deployment.

.. _grafana-connect:

Connect Grafana to CrateDB Cloud
================================

After setting up and logging into Grafana, you should be greeted by
Grafana Home page.

.. image:: _assets/img/grafana-welcome.png
   :alt: Grafana Home page

To visualize the data, you must add a data source. To do this, click on the
cogwheel "Settings" icon in the left menu bar. This should take you to the Data
sources Configuration page. 

.. image:: _assets/img/grafana-settings.png
   :alt: Grafana Settings

Once there, click on the *Add data source* button. Here, look up and choose
"PostgreSQL".

.. image:: _assets/img/grafana-search.png
   :alt: Grafana integrations

Once "PostgreSQL" is chosen, you will be brought to a form that you must fill

> out to connect to the CrateDB Cloud. Filled out example might look like the

Reviewer 2: ```suggestion
A filled out example might look like the
screenshot below
```

or maybe better would be

```suggestion
A completed example might look like the
screenshot below
```

screenshot below.

.. image:: _assets/img/grafana-connection.png
   :alt: Grafana connection form

> What can be different for you is *host* and *user* credentials. The host can be

Reviewer 2: ```suggestion
The *host* and *user* credentials may appear differently for you. The host can be
```

   is that your query needs to return a column called "time". Grafana will 
   identify this as your time metric, so make sure the column has the proper 
   datatype (any datatype representing an `epoch time`_). In this query, we're 
   labelling pickup_datetime as "time" for this reason.

Once you input these SQL statements, there are a couple of adjustments you can
make:

- On the top of the panel, select the appropriate time range for your
  panel—in this case, from July 1st to July 7th, 2019:

- Under "Settings" on the right, define the name of your panel.

- Under "Display", select "Bars".

After that, you should get a panel similar to this:

.. image:: _assets/img/grafana-panel1.png
   :alt: Grafana panel 1

When you're satisfied with the look of the panel, click *Apply*. This will
bring you back to the overview of the dashboard. Now it will have 1 panel
created in it. Click on the *Add panel* in the top menu bar and you can create
another one.

.. image:: _assets/img/grafana-add-panel.png
   :alt: Grafana add another panel to dashboard

Another question worth asking might be: What was the average distance per ride

> per day? To find that out, input this to the SQL console of the new panel:

Reviewer 2: ```suggestion
per day? To find this out, input the following SQL command into console of the new panel:
```
